### DISCUSSION TITLE
How to get the previous event that targets this state when inside of an action of a state with always eventless transitions?

### DISCUSSION THREAD

**Discussion by Norlandz** [statelyai/xstate#4542] [Q&A] [answered]
How to get the previous event that targets this state when inside of an action of a state with always eventless transitions?
<=>
How to get 
_ the previous event `evx__goOffline` (that targets this state `stage__offlineCleanup`) _
when inside of _ an action of a state `stage__offlineCleanup` (with always eventless transitions) _ ?

- (btw, for the naming, `stage <~=> state` in my language/understanding... just to avoid conflict naming with React state...)

```ts
    states: {
      stage__online: {
        on: {
          evx__goOffline: 'stage__offlineCleanup',
        },
      },

      stage__offlineCleanup: {
        always: {
          target: 'stage__offline',
          actions: (context, event, _meta) => {
            console.log(event); // << there is nothing inside this event?
            // how can I get the previous event `evx__goOffline` that targets this `stage__offlineCleanup`?
            // (in GuardMeta, it seems like I can get it through `_meta.state.event.type`? but not for ActionMeta?)
          },
        },
      },

      stage__offline: {
        type: 'final',
      },
    },
```

**Comment by Andarist** [comment, marked as answer]
We just released a new major version of XState. It solves this problem since we removed the notion of null events there. So `actions` in always transitions should be always executed with the `event` that triggered the original transition
